lib/index.js

```diff
@@ -888,6 +885,301 @@ module.exports = () => {
+        const pointPtr = charStar(point);
+        const ret = Module.ccall(
+          'ec_point_compress',
+          'number',
+          ['number', 'number', 'number', 'number'],
+          [output, outputLenPtr, pointPtr, point.length]
+        );
+
+        if (ret === 1) {
+          const res = new Uint8Array(
+            Module.HEAPU8.subarray(output, output + outputLen)
+          );
+          freeMalloc();
+          return res;
+        }
+        freeMalloc();
+        throw new Error('point_compress', ret);
+      }
+
+      /**
+       * @summary check if a point is private
+       * @arg {Uint8Array} point - point to check
+       * @returns {boolean} true if private, false otherwise
+       */
+      function isPrivate(point) {
+        if (!point || !(point instanceof Uint8Array)) {
+          throw new TypeError('d must be a Uint8Array');
+        }
+
+        const dPtr = charStar(point);
+        const ret = Module.ccall(
+          'ec_seckey_verify',
+          'number',
+          ['number'],
+          [dPtr]
+        );
+        freeMalloc();
+        return ret === 1;
+      }
+
+      /**
+       * @summary create point from scalar
+       * @arg {Uint8Array} scalar - scalar to check if on the curve
+       * @returns {Uint8Array | null} point if valid, null otherwise
+       */
+      // eslint-disable-next-line no-unused-vars
+      function pointFromScalar(scalar, _) {
+        if (!scalar || !(scalar instanceof Uint8Array)) {
+          throw new TypeError('scalar must be a Uint8Array');
+        }
+
+        const outputlen = 33;
+        const output = malloc(outputlen);
+        const outputLenPtr = malloc(4);
+        Module.setValue(outputLenPtr, outputlen, 'i32');
+
+        const ret = Module.ccall(
+          'ec_point_from_scalar',
+          'number',
+          ['number', 'number', 'number'],
+          [output, outputLenPtr, charStar(scalar)]
+        );
+        if (ret === 1) {
+          const res = new Uint8Array(
+            Module.HEAPU8.subarray(output, output + outputlen)
+          );
+          freeMalloc();
+          return res;
+        }
+        freeMalloc();
+        throw new Error('point_from_scalar', ret);
+      }
+
+      /**
+       * @summary add tweak to an x-only point
+       * @arg {Uint8Array} point - x-only point
+       * @arg {Uint8Array} tweak - tweak to add
+       * @return {Object} x-only point with tweak added and parity
+       */
+      function xOnlyPointAddTweak(point, tweak) {
+        if (!point || !(point instanceof Uint8Array) || point.length !== 32) {
+          throw new TypeError('p must be a Uint8Array of 32 bytes');
+        }
+        if (!tweak || !(tweak instanceof Uint8Array) || tweak.length !== 32) {
+          throw new TypeError('tweak must be a Uint8Array of 32 bytes');
+        }
+        const output = malloc(32);
+        const pPtr = charStar(point);
+        const tweakPtr = charStar(tweak);
+        const parityPtr = malloc(1);
```

Comment:
the parity arg is declared as `int` and since we know the values it can assume, we can allocate 4 bytes, which is the min number of bytes for an int in C:

```suggestion
        const parityPtr = malloc(4);
```